Core/PaymentMethodHelper.php

```diff
@@ -80,7 +82,143 @@ public static function getModulePayments()
      */
     public static function getMandate($iOrderNumber)
     {
-        $iLength = self::MAX_MANDATE_ID_LENGTH - 1 - strlen((string) time());
-        return new Mandate(substr($iOrderNumber, 0, $iLength) . '-' . time());
+        $sTime = (string) time();
+        $iLength = self::MAX_MANDATE_ID_LENGTH - 1 - strlen($sTime);
+        return new Mandate(substr($iOrderNumber, 0, $iLength) . '-' . $sTime);
+    }
+
+    /**
+     * Returns account holder for SEPA Direct Debit
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getAccountHolder()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['accountHolder'];
+    }
+
+
+    /**
+     * Returns IBAN
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getIban()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['iban'];
+    }
+
+    /**
+     * Returns BIC
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public function getBic()
+    {
+        $oSession = Registry::getConfig()->getSession();
+        $aDynvalues = $oSession->getVariable('dynvalue');
+        return $aDynvalues['bic'];
+    }
+
+    /**
+     * Generates SEPA mandate html body
+     *
+     * @param Basket $oBasket
+     *
+     * @return string
+     *
+     * @since 1.0.1
+     */
+    public static function getSepaMandateHtml($oBasket)
+    {
+        $oShop = Helper::getShop();
+        $iOrderNumber = Helper::getSessionChallenge();
```

Comment:
It is not very intuitive that `getSessionChallenge` is returning an order number, is it?